go/glome/glome.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2020 Google LLCDWE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package glome implements GLOME protocol.
+package glome
+
+import (
+	"fmt"
+	"io"
+
+	"crypto/hmac"
+	"crypto/sha256"
+
+	"golang.org/x/crypto/curve25519"
+)
+
+const (
+	// PrivateKeySize is the size of a PrivateKey in bytes.
+	PrivateKeySize = 32
+	// PublicKeySize is the size of a PublicKey in bytes.
+	PublicKeySize = 32
+	// MaxTagSize is the maximum size allowed for a Tag
+	MaxTagSize = 32
+	// MinTagSize is the minimum size allowed for a Tag
+	MinTagSize = 1
+)
+
+var (
+	// ErrInvalidPublicKey denotes that a slice that intend to be a public key is not of desired length
+	ErrInvalidPublicKey = fmt.Errorf("Invalid public key - byte slice len is not %d", PublicKeySize)
+	// ErrInvalidPrivateKey denotes that a slice that intend to be a private key is not of desired length
+	ErrInvalidPrivateKey = fmt.Errorf("Invalid private key - byte slice len is not %d", PrivateKeySize)
+	// ErrInvalidTagSize denotes that provided integer is not suitable to be minPeerTagSize
+	ErrInvalidTagSize = fmt.Errorf("Invalid tag size - minPeerTagSize must be in range %d-%d",
+		MinTagSize, MaxTagSize)
+	// ErrInvalidReader denotes that library failed to read PrivateKeySize bytes from given Reader.
+	ErrInvalidReader = fmt.Errorf("Invalid Reader - Failed to read  %d bytes", PrivateKeySize)
+)
+
+// PublicKey is the type of GLOME public Keys.
+//
+// It can be initializes either by casting a [PublicKeySize]byte array or from a byte
+// slice with the PublicKeyFromSlice function. For example:
+//
+//     //Generate Private Key as existing byte array
+//     b := [32]byte{0,2,...,7,6}
+//     p := glome.PublicKey(b)
+//     //Generate from byte slice
+//     s := b[:]
+//     p, _ := glome.PublicKeyFromSlice(s)
+type PublicKey [PublicKeySize]byte
+
+// PublicKeyFromSlice generates a PublicKey object from slice. Return ErrInvalidPublicKey
+// if slice's length is not PublicKeySize.
+func PublicKeyFromSlice(b []byte) (*PublicKey, error) {
+	if len(b) != PublicKeySize {
+		return nil, ErrInvalidPublicKey
+	}
+
+	var p PublicKey
+	copy(p[:], b)
+	return &p, nil
+}
+
+// PrivateKey is the type of GLOME public Keys.
+//
+// It can be initializes either by casting a [PrivateKeySize]byte array or from a byte
```

Comment:
```suggestion
// It can be initialized either by casting a [PrivateKeySize]byte array or from a byte
```

Reply:
This seems not to be resolved yet, and there is another instance of this typo further up.

Reply:
True, fixed!